astropy/table/tests/test_groups.py

```diff
@@ -636,9 +636,7 @@ def test_column_filter():
     """
 
     def all_positive(column):
-        if np.any(column < 0):
-            return False
-        return True
+        return not np.any(column < 0)
```

Comment:
A rewrite of the code should be
```suggestion
        return np.all(column >= 0)
```
as long as we don't have to worry about `nan` (it should be mentioned in a comment if that is a concern). But looking at the function name the correct code would actually be
```suggestion
        return np.all(column > 0)  # no NaN to worry about
```
or
```suggestion
        return not np.any(column <= 0)  # can handle NaN
```

Reply:
I'll go with your first suggestion as the other ones are not strictly equivalent.
Some definitions of "positive" include 0 so I'm not willing to bet the current behavior is buggy.